library/src/main/java/org/apache/apex/malhar/lib/state/managed/IncrementalCheckpointManager.java

```diff
@@ -109,8 +110,8 @@ public void run()
           transferWindowFiles();
           if (latestExpiredTimeBucket.get() > -1) {
             try {
-              managedStateContext.getBucketsFileSystem().deleteTimeBucketsLessThanEqualTo(
-                  latestExpiredTimeBucket.getAndSet(-1));
+              latestPurgedTimeBucket = latestExpiredTimeBucket.getAndSet(-1);
```

Comment:
Oh wait, I see that you're referring to the recorded latest purged time bucket. In that case, the latest purged time bucket will be reverted to the previous value but the time bucket was actually purged forever before the checkpoint. However, that should not matter since we will never need the purged time bucket anyway.